lib/Backend/Lower.cpp

```diff
@@ -26094,6 +26094,16 @@ Lowerer::GenerateGetImmutableOrScriptUnreferencedString(IR::RegOpnd * strOpnd, I
     {
         this->m_lowererMD.GenerateObjectTest(strOpnd, insertBeforeInstr, doneLabel);
     }
+
+    if (loweringCloneStr && func->IsLoopBody())
```

Comment:
>if (loweringCloneStr && func->IsLoopBody()) [](start = 4, length = 43)

LouisLaf(comment from other PR)
I expected your change to be in the runtime side instead of the JIT'd code. If we leave it in the JIT'd code, we should only do it for JIT loop bodies, but I kind of think it's just better to put this check in the helper itself. It minimizes the amount of code the JIT has to generate, which is better throughput wise and memory wise.

  @atulkatti
As discussed offline, we do have a vtable check to avoid the helper call when not necessary that needs this to be in JIT'd code. I will add a check to do this only for JIT loop bodies.